src/entities/Proposal/jobs.ts

```diff
@@ -91,41 +108,49 @@ export async function finishProposal(context: JobContext) {
   }
 
   context.log(`Updating ${pendingProposals.length} proposals...`)
-  const finishedProposals: ProposalAttributes[] = []
-  const acceptedProposals: ProposalAttributes[] = []
-  const rejectedProposals: ProposalAttributes[] = []
+  const finishedProposals: ProposalWithWinnerChoice[] = []
+  const acceptedProposals: ProposalWithWinnerChoice[] = []
+  const rejectedProposals: ProposalWithWinnerChoice[] = []
 
   for (const proposal of pendingProposals) {
     const outcome = await calculateOutcome(proposal, context)
-    switch (outcome) {
+    switch (outcome?.status) {
       case ProposalOutcome.REJECTED:
-        rejectedProposals.push(proposal)
+        rejectedProposals.push({ proposal, winnerChoice: outcome.winnerChoice })
```

Comment:
I think you can add this object to a variable `const proposalWithWinnerChoice = { proposal, winnerChoice: outcome.winnerChoice }` as you are using the same object in all cases.